test/second_q/hamiltonians/lattices/test_hexagonal_lattice.py

```diff
@@ -0,0 +1,133 @@
+                (3, 4, 1.42j),
+                (4, 5, 1.42j),
+                (5, 6, 1.42j),
+                (7, 8, 1.42j),
+                (8, 9, 1.42j),
+                (0, 3, 1.42j),
+                (2, 5, 1.42j),
+                (4, 7, 1.42j),
+                (6, 9, 1.42j),
+                (0, 0, 1.0),
+                (1, 1, 1.0),
+                (2, 2, 1.0),
+                (3, 3, 1.0),
+                (4, 4, 1.0),
+                (5, 5, 1.0),
+                (6, 6, 1.0),
+                (7, 7, 1.0),
+                (8, 8, 1.0),
+                (9, 9, 1.0),
+            ]
+            target_graph.add_edges_from(weighted_edge_list)
+            self.assertTrue(
+                is_isomorphic(hexa.graph, target_graph, edge_matcher=lambda x, y: x == y)
+            )
+
+        with self.subTest("Check the number of nodes."):
+            self.assertEqual(hexa.num_nodes, 10)
+
+        with self.subTest("Check the set of nodes."):
+            self.assertSetEqual(set(hexa.node_indexes), set(range(10)))
+
+        with self.subTest("Check the set of weights."):
+            target_set = {
+                (0, 1, 1.42j),
+                (1, 2, 1.42j),
+                (3, 4, 1.42j),
+                (4, 5, 1.42j),
+                (5, 6, 1.42j),
+                (7, 8, 1.42j),
+                (8, 9, 1.42j),
+                (0, 3, 1.42j),
+                (2, 5, 1.42j),
+                (4, 7, 1.42j),
+                (6, 9, 1.42j),
+                (0, 0, 1.0),
+                (1, 1, 1.0),
+                (2, 2, 1.0),
+                (3, 3, 1.0),
+                (4, 4, 1.0),
+                (5, 5, 1.0),
+                (6, 6, 1.0),
+                (7, 7, 1.0),
+                (8, 8, 1.0),
+                (9, 9, 1.0),
+            }
+            self.assertSetEqual(set(hexa.weighted_edge_list), target_set)
+
+        with self.subTest("Check the adjacency matrix."):
+            target_matrix = np.zeros((10, 10), dtype=complex)
+            np.fill_diagonal(target_matrix, 1.0)
+
+            neg_indices = [
+                (1, 0),
+                (2, 1),
+                (3, 0),
+                (4, 3),
+                (5, 2),
+                (5, 4),
+                (6, 5),
+                (7, 4),
+                (8, 7),
+                (9, 6),
+                (9, 8),
+            ]
+            pos_indices = [
+                (0, 1),
+                (0, 3),
+                (1, 2),
+                (2, 5),
+                (3, 4),
+                (4, 5),
+                (4, 7),
+                (5, 6),
+                (6, 9),
+                (7, 8),
+                (8, 9),
+            ]
+
+            for idx1, idx2 in neg_indices:
+                target_matrix[idx1, idx2] = 0 - 1.42j
+
+            for idx3, idx4 in pos_indices:
+                target_matrix[idx3, idx4] = 0 + 1.42j
+
+            assert_array_equal(hexa.to_adjacency_matrix(weighted=True), target_matrix)
```

Comment:
```suggestion
        with self.subTest("Check the adjacency matrix."):
            target_matrix = np.zeros((10, 10), dtype=complex)

            indices = [
                (0, 1),
                (0, 3),
                (1, 2),
                (2, 5),
                (3, 4),
                (4, 5),
                (4, 7),
                (5, 6),
                (6, 9),
                (7, 8),
                (8, 9),
            ]

            for idx1, idx2 in indices:
                target_matrix[idx1, idx2] = 0 + 1.42j

            target_matrix -= target_matrix.T

            np.fill_diagonal(target_matrix, 1.0)

            assert_array_equal(hexa.to_adjacency_matrix(weighted=True), target_matrix)
```
This is just a little bit shorter and it avoids storing the redundant information in the `neg_` and `pos_indices`